### ISSUE TITLE
Deprioritize rates without gas fees 

### ISSUE THREAD

**Issue by gomesalexandre** [shapeshift/web#8143]
### Overview

Raised by @0xApotheosis in https://github.com/shapeshift/web/pull/8079#pullrequestreview-2444196556:

> One thought on this whilst testing: in cases where we don't have a gas estimate on rate fetch (e.g. for Portals), in this branch we unfairly list that swapper above other swappers that do have a gas estimate.

> In the example below LiFi is actually a much better rate overall (very low gas vs what Portals will end up being), but it ranks below Portals.

And indeed, not *knowing* network fees (as in, *undefined* unknown network fees vs. 0 explicit network fees e.g CoW) means the gas could make a given rate a lot worse for users in the end.

This PR is to ensure we deprioritize rates with unknown gas estimates. 

### References and additional details

 https://github.com/shapeshift/web/pull/8079

### Acceptance Criteria

- Rates with `undefined` network fees are deprioritized in profit of rates which *may* seem better at first glance but may be worse given unknown fees i.e Portals atm

**Comment by 0xean** [comment]
portals for the moment.

**Comment by twblack88** [comment]
yes we now return gas estimates (like 812,423 gas) but not the actual gas price stuff (like base fee, max fee etc. as this is often wallet dependent) 

`outputToken": "0xc5dc397b1db51da30dc9f3ac7084bbba1efbe249",
    "outputAmount": "59788",
    "minOutputAmount": "59639",
    "outputTokenDecimals": 8,
    "context": {
        "inputToken": "base:0x0000000000000000000000000000000000000000",
        "inputAmount": "22350000000000000",
        "inputAmountUsd": 58.2443235,
        "outputToken": "base:0xc5dc397b1db51da30dc9f3ac7084bbba1efbe249",
        "outputAmount": "59788",
        "outputAmountUsd": 58.18292382492995,
        "minOutputAmountUsd": 58.03792389768846,
        "slippageTolerancePercentage": 0.3554,
        "gasLimit": "871532",
        "simulated": true
    }
`
